web/src/components/User/Login/Login.tsx

```diff
@@ -1,192 +1,252 @@
+import jwt_decode from 'jwt-decode'
+import { useContext, useEffect, useState } from 'react'
 import LoginAPI from '../../../requests/LoginAPI'
-import About from './About'
-import Cookie from './Cookie'
+
+import type { IJWT_Token, IUser } from '../../../utils/interfaces/interfaces'
+
+import { UserContext } from '../../../contexts/User.context'
+import { useCooldown } from '../../../utils/hooks/useCooldown'
+
 import { useNavigate } from 'react-router-dom'
-import jwt_decode from 'jwt-decode'
-import { verifyEmailAddress } from '../../../requests/Login-API/verify-email-address'
-import { checkValidToken, saveDataToLocalStorage, DeleteLocalStorage } from './LoginHelper'
-import { JWT_Token } from '../../../utils/interfaces/interfaces'
 
 //install google login package by running: npm install @react-oauth/google@latest
 //install jwt-decode by running: npm install jwt-decode
 
 // This is the client ID of the Google OAuth app
-const clientId = '250149314571-cfinl9pkdvrv7epjvmid5uqve75ohk48.apps.googleusercontent.com'
+const CLIENT_ID =
+    '250149314571-cfinl9pkdvrv7epjvmid5uqve75ohk48.apps.googleusercontent.com'
 
 const Login = () => {
-    const countDownInit = 60
-    const [countDownCurr, setCountDownCurr] = useState(countDownInit)
-    const [isOneTapHidden, setOnetapHidden] = useState(true)
-    const [showSpan, setShowSpan] = useState(false)
-    const [isButtonHidden, setButtonHidden] = useState(false)
-    const [emailOriginalInput, setEmailOriginalInput] = useState('?')
-    const [emailInput, setEmailInput] = useState('?')
-    const [codeInput, setCodeInput] = useState('??')
-
+    const [email_address, set_email_address] = useState('')
+    const [auth_code, set_auth_code] = useState('')
+    const { user, set_user } = useContext(UserContext)
+    const [auth_cooldown, start_cooldown] = useCooldown(60)
     const navigate = useNavigate()
 
-    // detect if token is already stored
-    // if yes, then navigate to dashboard page
-    if (checkValidToken()) {
-        navigate("/dashboard");
+    /** custom navigation fx, add animation before navigation
+     */
+    const navigate_to_main_page = () => {
+        console.log(user)
+        setTimeout(() => {
+            set_sign_in_button_msg('\u2713')
+        }, 300)
+
+        setTimeout(() => {
+            navigate('/dashboard')
+        }, 1500)
     }
 
+    /** try fetch user from localstorage/context on init
+     */
+    useEffect(() => {
+        const storedUser: IUser = JSON.parse(
+            sessionStorage.getItem('user') ?? '{}'
+        )
 
-    function errorAlert(msg: SetStateAction<string>) {
-        let des = document.getElementById('login-description')
-        if (des != undefined) {
-            des.className = 'text-yellow-300 text-bold text-[18px]'
-            des.textContent = '\u26A0' + '\n' + msg
+        if ('email' in storedUser) {
+            set_user(storedUser)
+            navigate_to_main_page()
```

Comment:
minor thing, we could remove the navigate function here? Since the following if clause will catch this case anyways.

Reply:
fixed